server/hap2/hatohol/hap2_nagios_livestatus.py

```diff
@@ -0,0 +1,311 @@
+        STATE_OK: "GOOD", STATE_WARNING: "BAD", STATE_CRITICAL: "BAD"}
+
+    DEFAULT_PATH = "/tmp/nagios.sock"
+    INITIAL_LAST_INFO = ""
+
+    def __init__(self):
+        self.__access_point = self.DEFAULT_PATH
+        self.__port = None
+        self.__time_offset = datetime.timedelta(seconds=time.timezone)
+        self.__trigger_last_info = None
+        self.__latest_statehist = None
+
+    def ensure_connection(self):
+        # load MonitoringServerInfo
+        ms_info = self.get_ms_info()
+        if ms_info is None:
+            logger.info("Use default connection parameters.")
+        else:
+            self.__access_point, self.__port = self.__parse_url(ms_info.url)
+
+        logger.info("Try to connection: Sv: %s" % self.__access_point)
+
+        socket_arg = self.__access_point
+        try:
+            if self.__port: socket_arg = (socket_arg, self.__port)
+            self.__socket = Socket(socket_arg)
+            test_query = self.__socket.test
+            test_query.call()
+        except socket.error as (errno, msg):
+            logger.error('Socket Error [%d]: %s' % (errno, msg))
+            raise hap.Signal
+
+    def __parse_url(self, url):
+        # [URL] "SERVER_IP:PORT" or "PATH"
+        colon_idx = url.find(":")
+        if colon_idx == -1:
+            path = url
+            return path, None
+        else:
+            port = url[colon_idx+1:]
+            server = url[0:colon_idx]
+            return server, port
+
+    def collect_hosts_and_put(self):
+        query = self.__socket.hosts.columns("name", "alias")
+        result = query.call()
+        hosts = [{"hostId": host["name"], "hostName": host["alias"]} for host in result]
+        self.put_hosts(hosts)
+
+    def collect_host_groups_and_put(self):
+        query = self.__socket.hostgroups.columns("name", "alias")
+        result = query.call()
+        groups = \
+          [{"groupId": group["name"], "groupName": group["alias"]} for group in result]
+        self.put_host_groups(groups)
+
+    def collect_host_group_membership_and_put(self):
+        query = self.__socket.hosts.columns("name", "groups")
+        result = query.call()
+        membership = [{"hostId": host["name"], "groupIds": host["groups"]} for host in result]
+
+        self.put_host_group_membership(membership)
+
+    def collect_triggers_and_put(self, fetch_id=None, host_ids=None):
+        query = self.__socket.services.columns("plugin_output",
+                                               "description",
+                                               "last_state_change",
+                                               "host_alias",
+                                               "host_name",
+                                               "state")
+
+        if host_ids is not None:
+            filter_condition = "host_name ~ "
+            for host_id in enumerate(host_ids):
+                if host_id[0] == 0: filter_condition += host_id[1]
```

Comment:
These two lines contains two statements in one line.
This condensed style is orthogonal against Python coding conventions.
We should do simple. One line should contain one statement.